src/msw/toolbar.cpp

```diff
@@ -1694,8 +1694,50 @@ bool wxToolBar::MSWOnNotify(int WXUNUSED(idCtrl),
                 nmtbcd->clrTextHighlight = wxColourToRGB(GetForegroundColour());
                 nmtbcd->clrHighlightHotTrack = wxSysColourToRGB(wxSYS_COLOUR_HOTLIGHT);
 
-                *result = CDRF_DODEFAULT | TBCDRF_USECDCOLORS | TBCDRF_HILITEHOTTRACK;
+                *result = CDRF_DODEFAULT | TBCDRF_USECDCOLORS | TBCDRF_HILITEHOTTRACK | CDRF_NOTIFYPOSTPAINT;
                 return true;
+
+            case CDDS_ITEMPOSTPAINT: {
+                // custom draw the drop-down arrow here, as it is always black
+                TBBUTTONINFO bi = { 0 };
+                bi.cbSize = sizeof(TBBUTTONINFO);
+                bi.dwMask = TBIF_STYLE | TBIF_COMMAND;
+                int itemIndex = (int) ::SendMessage(GetHwnd(), TB_GETBUTTONINFO, (WPARAM)nmtbcd->nmcd.dwItemSpec, (LPARAM)&bi);
+                if (itemIndex >= 0 && bi.fsStyle & TBSTYLE_DROPDOWN) {
+                    RECT ddrc = { 0 };
+                    ::SendMessage(GetHwnd(), TB_GETITEMDROPDOWNRECT, (WPARAM)itemIndex, (LPARAM)&ddrc);
+
+                    if (nmtbcd->nmcd.uItemState & CDIS_HOT) {
+                        HBRUSH bgBrush = CreateSolidBrush(wxColourToRGB(GetBackgroundColour().ChangeLightness(110)));
+                        ::FillRect(nmtbcd->nmcd.hdc, &ddrc, bgBrush);
+                        DeleteObject(bgBrush);
+                    }
+                    else {
+                        HBRUSH bgBrush = CreateSolidBrush(wxColourToRGB(GetBackgroundColour()));
+                        ::FillRect(nmtbcd->nmcd.hdc, &ddrc, bgBrush);
+                        DeleteObject(bgBrush);
+                    }
```

Comment:
It looks like we should merge with the `if` branch and just use a different colour here.

Reply:
No, not just the else case, otherwise you won't see any separation between the dropdown and the actual button when you hover over it with the mouse.